### ISSUE TITLE
xCode 14.3 and iOS 16.4 simulator crashing while running tests on it without rosetta support

### ISSUE THREAD

**Issue by Tirth21896** [nalexn/ViewInspector#245]
<img width="2382" alt="Screenshot 2023-05-12 at 3 16 29 PM" src="https://github.com/nalexn/ViewInspector/assets/26151224/b26f85b2-3a11-4822-b3e8-d0c2b1ea226e">
<img width="2390" alt="Screenshot 2023-05-12 at 3 16 51 PM" src="https://github.com/nalexn/ViewInspector/assets/26151224/e79dec7e-684b-4643-bf02-1cd63e0facd3">

We are using ViewInspector for a long now and we have been loving it. Recently my team migrated our codebase to Xcode 14.3 and we all know its not support rosetta for MAC M series machines we were forced to migrate our project to Xcode 14.3 where I noticed a thing in which ViewInspector was crashing and making it impossible to run our existing Unit Tests. So I investigated it below is what I found.

Given : 
- our project has ViewInspector Cocoapod 0.9.6
When : 
- we are under environment Xcode 14.3 and iOS simulator is on iOS 16.4 (without Rosetta) 
Then :
- it works just fine and the test passes and fails where it suppose to be just fine with a full run on the unit tests project as expected.

However under circumstances of 
Given :
- our project has ViewInspector Cocoapod 0.9.6
When : 
- we are under environment Xcode 14.3 and iOS simulator is on iOS 16.4 (with Rosetta) 
Then : 
- it is not able to do a full run of Unit Tests.
- it would fail where it should but it crashes where we have used environment objects -> these test passes just fine when running them on simulators with rosetta.
- we are stuck since we cant fully get rid of rosetta.

To summarize, We looked into our project and the way we are writing out tests using ViewInspector but we could not find a location, we could not find a direction towards which we can go to resolve this issue. and therefore I am creating this issue so in case if you can follow the above steps and see if you can regenerate this issue. 

Thanks.

**Comment by iNono22** [comment]
Having the same issue, it was working before but now that I'm not using Rosetta anymore it crashes at the same place.
I am beating my brains out to find a solution for few days now, no luck so far. Please help.

**Comment by AnthonyTTaylor** [comment]
Similar issue for myself as well. Has prevented myself from running my tests on my application while this bug persists. 
Upgraded to latest version of ViewInspector, and still crashes.

**Comment by sweetclimusic** [comment]
I was experiencing this as well... but only after adding a call to an `@Environment(\.sizeCategory) var sizeCategory` as a way to rescale text for accessibility.  anyway I just want to ask if any of you are also using @Environment in a custom view or view Modifier? You may have to delay the call to inspecting your view.  See here, I implemented method 1
https://github.com/nalexn/ViewInspector/blob/0.9.7/guide.md#views-using-state-environment-or-environmentobject

**Comment by Tirth21896** [comment]
In my case where the crash is happening - that particular view does have a requirement of @EnvironmentObject - however I am passing in the @EnvironmentObject.

- I did try what you suggested using Approch #1 in suggested link by you - but its crashing the Viewhosting.Host line as the view that I am passing inside is like below
`func testBody() throws {
        
        let exp = view.on(\.didAppear) { inspectable in
            XCTAssertNotNil(inspectable)

            let mainHStack = try inspectable.find(ViewType.HStack.self)

            let buttons = mainHStack.findAll(ViewType.Button.self)
            XCTAssertEqual(buttons.count, 2)

            let videoButton = try inspectable.find(viewWithAccessibilityIdentifier: self.videoIdentifier).button()
            XCTAssertEqual(try videoButton.labelView().text().string(), "Video")
            XCTAssertNoThrow(try videoButton.tap())

            let photoButton = try inspectable.find(viewWithAccessibilityIdentifier: self.photoIdentifier).button()
            XCTAssertEqual(try photoButton.labelView().text().string(), "1")
            XCTAssertNoThrow(try photoButton.tap())
        }
        ViewHosting.host(view: view.environmentObject(vehicleViewModel))
        wait(for: [exp], timeout: 15.0)
    }`

- the original implementation that I had was using Approch #2

**Comment by nalexn** [comment]
Could you guys run the ViewInspector projects' tests with your setup and see if any test crashes? I've tried on the 16.4 simulator with and without rosetta, and the tests pass ok.
I need a way to repro this - please try to locate the problematic view in your project and share the minimum code that reproduces the crash.

**Comment by mrk-han** [comment]
Agree, a minimal sample project here would be crucial for debugging the core issue

**Comment by Tirth21896** [comment]
Hey you all,

I was trying to create a scenario where I could replicate this issue, however with a new project and importing ViewInspector using Cocoapods I am not able to repro it; so I think for the good of this thread - we can deactivate it for now.
